Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -2098,13 +2121,82 @@ private bool CleanTempDirectory(AbsolutePath tempDirectoryPath)
             return true;
         }
 
-        private void LogTempDirectoryTooLong(string directory)
+        private bool PrepareTempDirectoryForVm(AbsolutePath tempDirectoryPath)
         {
-            Tracing.Logger.Log.PipProcessTempDirectoryTooLong(
+            // Suppose that the temp directory is D:\Bxl\Out\Object\Pip123\Temp\t_0.
+            string path = tempDirectoryPath.ToString(m_pathTable);
+
+            // Delete any existence of path D:\Bxl\Out\Object\Pip123\Temp\t_0.
+            try
+            {
+                if (FileUtilities.FileExistsNoFollow(path))
+                {
+                    // Path exists as a file or a symlink (directory/file symlink).
+                    FileUtilities.DeleteFile(path, waitUntilDeletionFinished: true, tempDirectoryCleaner: m_tempDirectoryCleaner);
+                }
+
+                if (FileUtilities.DirectoryExistsNoFollow(path))
+                {
+                    // Path exists as a real directory: wipe out that directory.
+                    FileUtilities.DeleteDirectoryContents(path, deleteRootDirectory: true, tempDirectoryCleaner: m_tempDirectoryCleaner);
+                }

```

Comment:
Why not call PreparePathForDirectory?

Reply:
Because I need to wipe out the existence of the path. In `PreparePathForDirectory`, if the directory exists, then we just wipe out its contents, but not the directory itself.

---
In reply to: [288714193](https://github.com/microsoft/BuildXL/pull/382#discussion_r288714193) [](ancestors = 288714193)